tests/e2e/hcp_machine_pool_test.go

```diff
@@ -575,5 +577,170 @@ var _ = Describe("HCP Machine Pool", labels.Feature.Machinepool, func() {
 						ContainSubstring("The last node pool can not be deleted from a cluster"))
 			}
 		})
+
+		It("upgrade - [id:67419]", labels.Medium, labels.Runtime.Day2, func() {
+			var err error
+
+			clusterService := rosaClient.Cluster
+			machinePoolUpgradeService := rosaClient.MachinePoolUpgrade
+			versionService := rosaClient.Version
+
+			By("Retrieve cluster config")
+			clusterConfig, err = config.ParseClusterProfile()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("without machinepool")
+			_, err = machinePoolUpgradeService.CreateManualUpgrade(
+				clusterID,
+				"",
+				clusterConfig.Version.RawID,
+				"",
+				"")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("expected a valid identifier for the machine pool"))
+
+			By("with wrong machinepool")
+			_, err = machinePoolUpgradeService.CreateManualUpgrade(
+				clusterID,
+				"anything",
+				clusterConfig.Version.RawID,
+				"",
+				"")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("Machine pool 'anything' does not exist for hosted cluster"))
+
+			By("for not-ready cluster")
+			var notReadyClusterID string
+			rosaClient.Runner.UnsetArgs()
+			output, err := clusterService.List()
+			Expect(err).To(BeNil())
+			clusterList, err := clusterService.ReflectClusterList(output)
+			Expect(err).To(BeNil())
+			for _, c := range clusterList.Clusters {
+				if c.State != constants.Ready && c.Topology == constants.HostedCP {
+					notReadyClusterID = c.ID
+					break
+				}
+			}
+			if notReadyClusterID != "" {
+				_, err = machinePoolUpgradeService.CreateManualUpgrade(
+					notReadyClusterID,
+					"anything",
+					clusterConfig.Version.RawID,
+					"",
+					"")
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("Cluster '%s' is not yet ready", notReadyClusterID)))
+			} else {
+				Logger.Info("No cluster found in state not-ready. Skipping this step")
+			}
+
+			By("Find a version has available upgrade versions")
```

Comment:
For each of these blocks, would it make sense to have a single execution statement and that loops through passed in flags? 

Something similar to this: https://github.com/openshift/rosa/blob/3a9ac1fa489b6d6595f43338c2674af2d463a5ea/tests/e2e/test_rosacli_node_pool.go#L1150

Reply:
We are in this case testing different cli options and we use either `CreateAutomaticUpgrade` or `CreateManualUpgrade` method.
I would see it as not readable to have a huge map containing all cases.